### DISCUSSION TITLE
Make ESLint part of test:codebase

### DISCUSSION THREAD

**Discussion by tcmetzger** [bokeh/bokeh#11305] [Build / CI]
While talking about https://github.com/bokeh/bokeh/pull/11304, @bryevdv and I were wondering if it would make it easier for local development and testing to have ESLint included when calling `node make test:codebase` or  `node make test`. To have linting included with codebase tests might make things more consistent for new contributors who might expect linting to be part of codebase testing (and include that in a pre-push git hook, for example). A possible downside would be that this might increase the CI's run time.

**Comment by mattpap** [comment]
Linting is very slow, so this would negatively affect overall development experience. Given this idea is targeted towards newbies, I would actually enable linting in `node make` (`node make all`) (perhaps maybe `node make build`) instead. In fact it may be considered a bug that `node make all` doesn't currently do linting. However, speed could be improved with incremental linting (this would have to be implemented by us), so eventually this could make its way into codebase tests.
